tests/test_frame.py

```diff
@@ -1457,6 +1457,19 @@ def test_types_from_dict() -> None:
         },
         orient="tight",
     )
+    # added following #896
+    data = {"l1": [1, 2, 3], "l2": [4, 5, 6]}
+    # testing `dtype`
+    pd.DataFrame.from_dict(data, orient="index", dtype="float")
+    pd.DataFrame.from_dict(data, orient="index", dtype=float)
+    pd.DataFrame.from_dict(data, orient="index", dtype=None)
+    # testing `columns`
+    pd.DataFrame.from_dict(data, orient="index", columns=["a", "b", "c"])
+    pd.DataFrame.from_dict(
+        data, orient="index", columns=[1.0, 2, datetime.datetime.now()]
+    )
```

Comment:
Can you surround these with the `check(assert_type(...` pattern?

Reply:
corrected in 729bdb5f819cff22a3abd342741275cd839fd8e2